build_process/package-management.md

```diff
@@ -0,0 +1,109 @@
+Additional labels for pre-release and build metadata are available as extensions to the MAJOR.MINOR.PATCH format.
+
+Proper versioning is extremely valuable as an asynchronous communication channel. This is how we as publishers communicate what types of changes can be expected in each release of our software.
+
+Semver is commonly understood to be important in open-source work, but it is equally important for closed-source projects. Internally facing systems can be consumed by multiple teams and used in multiple projects. Establishing good versioning practices early on is critical to the success of our projects.
+
+It's important to know that semver is not perfect. Regardless of the strategies involved in incrementing versions (manual, [conventional commits][conventional], or anything else), new versions can only be as accurate as the people applying the version changes. Mistakes can happen. Lean on [npm deprecate][deprecate] when needed.
+
+## Creating
+New open source packages created under the Sparkbox organization should typically be [scoped][scope] to `@sparkbox`. This is not a hard rule, but it is highly encouraged.
+
+To quickly create a new package run `npm init -y --scope sparkbox`. This will create a `package.json` something like the following:
+```js
+{
+  "name": "@sparkbox/standard",
+  "version": "1.0.0",
+  "description": "",
+  "main": "index.js",
+  "scripts": {
+    "test": "echo \"Error: no test specified\" && exit 1"
+  },
+  "repository": {
+    "type": "git",
+    "url": "git+https://github.com/sparkbox/standard.git"
+  },
+  "keywords": [],
+  "author": "",
+  "license": "ISC",
+  "bugs": {
+    "url": "https://github.com/sparkbox/standard/issues"
+  },
+  "homepage": "https://github.com/sparkbox/standard#readme"
+}
+```
+
+Private packages will typically be used in client work (design systems, component libraries, various utilities). Consider scoping these to something company-appropriate to avoid naming collisions with open-source packages.
+
+Additional fields that we will most likely want to use in package.json:
+- [module][pkgModule] - offer an ES module entry point
+- [files][pkgFiles] - don't package more than we need
+- [engines][pkgEngines] - tell people and machines what node/npm version we expect
+
+Read more about [package.json][pkg] fields.
+
+For [initial versions][initialVersion], we should stick to the semver guidelines. Release a zero major (0.1.0) if we expect a lot of churn early on. This gives us time to establish our package's APIs before committing to a stable release. If we are comfortable with the stability of our initial release, then start at 1.0.0.
+
+## Publishing
+For continuous integration systems, we should [generate an auth token](https://docs.npmjs.com/creating-and-viewing-authentication-tokens) and read it from the environment in .npmrc before publishing.
+
+```sh
+$ echo "//registry.npmjs.org/:_authToken=${NPM_TOKEN}" > ./.npmrc
```

Comment:
Link to [12 factor app](https://12factor.net/config)

Reply:
done.